### PR TITLE
Document config_providers

### PR DESCRIPTION
**Pull request by fanny-jiang**
### What does this PR do?
Add new `Autodiscovery` section to the Kubernetes and Docker further configuration pages with environment variable descriptions for `DD_CONFIG_PROVIDERS`, `DD_EXTRA_CONFIG_PROVIDERS`, `DD_LISTENERS`, and `DD_EXTRA_LISTENERS`.

### Motivation
GH issue

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/containers/docker/_index.md [hunk 1]
```diff
@@ -229,15 +229,22 @@ Additional examples are available on the [Container Discover Management][25] pag
...
 
+### Autodiscovery
+
+| Env Variable                 | Description                                                                                                                                                                           |
+|------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `DD_LISTENERS`               | Autodiscovery listeners to run.                                                                                                                                                       |
+| `DD_EXTRA_LISTENERS`         | Additional Autodiscovery listeners to run. They are added in addition to the variables defined in the `listeners` section of the `datadog.yaml` configuration file.                   |
+| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. The default provider is `docker`. The Docker provider handles templates embedded in container labels. |
```
Comment (fanny-jiang) [inline content/en/containers/docker/_index.md:238]: Should the description for `DD_CONFIG_PROVIDERS` in this Docker-specific doc match the description in the Kubernetes-specific docs? It's the same configuration option and the supported values are the same, but in Docker, only the `docker` value is applicable.

Comment (cswatt) [reply content/en/containers/docker/_index.md:238]: let's keep what you have! i think it works very well!

File: content/en/containers/guide/kubernetes_daemonset.md [hunk 2]
```diff
@@ -260,6 +260,15 @@ Additional examples are available on the [Container Discover Management][27] pag
...
+| `DD_EXTRA_LISTENERS`         | Additional Autodiscovery listeners to run. They are added in addition to the variables defined in the `listeners` section of the `datadog.yaml` configuration file.                                                                                                                                                                                                                                                                                                                                    |
+| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. Available providers are: <br>`kubelet` - The kubelet provider handles templates embedded in pod annotations. <br>`docker` - The Docker provider handles templates embedded in container labels. <br> `clusterchecks` - The clustercheck provider retrieves cluster-level check configurations from the cluster-agent. <br>`kube_services` - The kube_services provider watches Kubernetes services for cluster-checks. |
```
Comment (cswatt) [inline content/en/containers/guide/kubernetes_daemonset.md:269]:
```suggestion
| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. Available providers are: <br>`kubelet` - Handles templates embedded in pod annotations. <br>`docker` - Handles templates embedded in container labels. <br> `clusterchecks` - Retrieves cluster-level check configurations from the Cluster Agent. <br>`kube_services` - Watches Kubernetes services for cluster checks. |
```

File: content/en/containers/kubernetes/configuration.md [hunk 3]
```diff
@@ -667,6 +667,16 @@ Starting with Agent v6.4.0 (and v6.5.0 for the Trace Agent), you can override th
...
+| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. Available providers are: <br>`kubelet` - The kubelet provider handles templates embedded in pod annotations. <br>`docker` - The Docker provider handles templates embedded in container labels. <br>`clusterchecks` - The clustercheck provider retrieves cluster-level check configurations from the cluster-agent. <br>`kube_services` - The kube_services provider watches Kubernetes services for cluster-checks. |
+| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. Available providers are: <br>`kubelet` - The kubelet provider handles templates embedded in pod annotations. <br>`docker` - The Docker provider handles templates embedded in container labels. <br>`clusterchecks` - The clustercheck provider retrieves cluster-level check configurations from the cluster-agent. <br>`kube_services` - The kube_services provider watches Kubernetes services for cluster-checks. |
```
Comment (cswatt) [inline content/en/containers/kubernetes/configuration.md:677]:
```suggestion
| `DD_CONFIG_PROVIDERS`        | The providers the Agent should call to collect checks configurations. Available providers are: <br>`kubelet` - Handles templates embedded in pod annotations. <br>`docker` - Handles templates embedded in container labels. <br>`clusterchecks` - Retrieves cluster-level check configurations from the Cluster Agent. <br>`kube_services` - Watches Kubernetes services for cluster checks. |
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**